src/runtime_src/tools/xclbinutil/unittests/FileCheck/FileCheck.py

```diff
@@ -0,0 +1,84 @@
+from argparse import RawDescriptionHelpFormatter
+import argparse
+import os
+import subprocess
+
+# Start of our unit test
+# -- main() -------------------------------------------------------------------
+#
+# The entry point to this script.
+#
+# Note: It is called at the end of this script so that the other functions
+#       and classes have been defined and the syntax validated
+def main():
+  # -- Configure the argument parser
+  parser = argparse.ArgumentParser(formatter_class=RawDescriptionHelpFormatter, description='description:\n  Unit test wrapper for the AIE TRACE METADATA section')
+  parser.add_argument('--resource-dir', nargs='?', default=".", help='directory containing data to be used by this unit test')
+  args = parser.parse_args()
+
+  # Validate that the resource directory is valid
+  if not os.path.exists(args.resource_dir):
+      raise Exception("Error: The resource-dir '" + args.resource_dir +"' does not exist")
+
+  if not os.path.isdir(args.resource_dir):
+      raise Exception("Error: The resource-dir '" + args.resource_dir +"' is not a directory")
+
+  # Prepare for testing
+  xclbinutil = "xclbinutil"
+
+  # Start the tests
+  print ("Starting test")
+
+  # ---------------------------------------------------------------------------
+
+  step = "1) Use the file check on a valid xclbin that has platformVBNV information and atleast one section"
+
```

Comment:
I think it is better to do it in two steps
1. create xcxlbin
2. run file-check against the xclbin (since this is what user would most likely do)

Stretch:
create a xclbin without PlatformVBNV
run file-check to see it actually returns the error message you expect

Reply:
Done. Added negative cases when the DRC would fail and validated the error message for eg: 1) xclbin with only one section 2)  xclbin with only platformVBnv information